### PR TITLE
breakout box: add worker sample

### PR DESCRIPTION
**Pull request by fippo**
I've been trying to be as minimal as possible. @guidou @dogben can you take a look?

### CODE REVIEW

**Review by guidou — APPROVED** [review]

File: src/content/insertable-streams/video-crop/index.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    <video id="localVideo" playsinline autoplay muted></video>
+    <video id="croppedVideo" playsinline autoplay muted></video>
+
+    <div class="box">
+        <button id="startButton">Start</button>
+    </div>
+
+    <p>
+        <b>Note</b>: This sample is using an experimental API that has not yet been standardized. As
+        of 2021-07-16, this API is available in Chrome M88 if the experimental code is enabled on
```
Comment (guidou) [inline src/content/insertable-streams/video-crop/index.html]: Will this work as early as 88?
Probably better to use  a more recent version where the API was actually working (maybe 90 or 91)

**Review by guidou — APPROVED** [review]
lgtm % comment

File: src/content/insertable-streams/video-crop/index.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    <meta name="viewport" content="width=device-width, user-scalable=yes, initial-scale=1, maximum-scale=1">
+    <meta itemprop="description" content="Client-side WebRTC code samples">
+    <meta itemprop="image" content="../../../images/webrtc-icon-192x192.png">
+    <meta itemprop="name" content="WebRTC code samples">
+    <meta name="mobile-web-app-capable" content="yes">
+    <meta id="theme-color" name="theme-color" content="#ffffff">
+
+    <base target="_blank">
+
+    <title>Peer connection</title>
```
Comment (dogben) [inline src/content/insertable-streams/video-crop/index.html]: Looks like the wrong title.

File: src/content/insertable-streams/video-crop/index.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+</head>
+
+<body>
+
+<div id="container">
+    <h1><a href="//webrtc.github.io/samples/" title="WebRTC samples homepage">WebRTC samples</a>
+        <span>Breakout Box crop</span></h1>
+
+    <p>This sample shows how to perform cropping on a video stream using the experimental
+        <a href="https://github.com/w3c/mediacapture-insertable-streams">insertable streams</a> API
```
Comment (dogben) [inline src/content/insertable-streams/video-crop/index.html]: nit: https://github.com/w3c/mediacapture-transform

File: src/content/insertable-streams/video-crop/js/main.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+const localVideo = document.getElementById('localVideo');
+const croppedVideo = document.getElementById('croppedVideo');
+
+const worker = new Worker('./js/worker.js', {name: 'Crop worker'});
+startButton.addEventListener('click', async () => {
+  const stream = await navigator.mediaDevices.getUserMedia({video: {width: 1280, height: 720}});
+  localVideo.srcObject = stream;
+
+  const [track] = stream.getTracks();
+  const processor = new MediaStreamTrackProcessor(track);
```
Comment (dogben) [inline src/content/insertable-streams/video-crop/js/main.js]: nit: Looks like the deprecated API. See https://w3c.github.io/mediacapture-transform/#track-processor-interface

File: src/content/insertable-streams/video-crop/js/main.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  const {readable} = processor;
+
+  const generator = new MediaStreamTrackGenerator('video');
```
Comment (dogben) [inline src/content/insertable-streams/video-crop/js/main.js]: nit: Ditto. See https://w3c.github.io/mediacapture-transform/#generator-interface

File: src/content/insertable-streams/video-crop/js/worker.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+const canvas = new OffscreenCanvas(640, 360);
+const ctx = canvas.getContext('2d', {desynchronized: true});
+
+async function transform(frame, controller) {
+  // Cropping is a bit too complex. See https://github.com/w3c/webcodecs/issues/281
+  ctx.drawImage(frame,
+      320, 180,
+      640, 360,
+      0, 0, 640, 360);
```
Comment (dogben) [inline src/content/insertable-streams/video-crop/js/worker.js:20]: nit: scale params aren't really needed if you're not scaling

File: src/content/insertable-streams/video-crop/js/worker.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+
+const canvas = new OffscreenCanvas(640, 360);
+const ctx = canvas.getContext('2d', {desynchronized: true});
+
+async function transform(frame, controller) {
```
Comment (dogben) [inline src/content/insertable-streams/video-crop/js/worker.js]: nit: looks like this doesn't need to be async

**Review by dogben — APPROVED** [review]
Sorry for the delay; I didn't see this when it was sent. You should probably fix the title, otherwise just ignorable comments.

**Comment by dogben** [comment]
I should also mention that my intention is to update https://webrtc.github.io/samples/src/content/insertable-streams/video-processing/ with worker processing, but I haven't had the chance yet.

File: src/content/insertable-streams/video-crop/js/worker.js (hunk 6 above)
Comment (fippo) [reply src/content/insertable-streams/video-crop/js/worker.js:20]: oddly omitting them led to a overload resolution error...

**Comment by fippo** [comment]
hah, thanks for catching the bad style!
I don't think the video processing sample necessarily requires a worker, having at least non-worker one is good. Also its complex enough already :-)

File: src/content/insertable-streams/video-crop/js/worker.js (hunk 6 above)
Comment (dogben) [reply src/content/insertable-streams/video-crop/js/worker.js:20]: Sorry, my mistake; the 5-parameter version does scaling, not cropping.

Comment (fippo) [reply src/content/insertable-streams/video-crop/js/worker.js:20]: the overload behaviour is clearly a mess hence the doc link :-)

**Pull request merged**